### PR TITLE
tiered: add typescript support

### PR DESCRIPTION
**Pull request by christianllv**
**Please check if the PR fulfills these requirements**
- [x] Followed the [Contributing](https://github.com/jaredwray/keyv/blob/main/CONTRIBUTING.md) guidelines.
- [x] Tests for the changes have been added (for bug fixes/features) with 100% code coverage.
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

### CODE REVIEW

File: packages/tiered/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
+import EventEmitter from 'node:events';
+import Keyv from 'keyv';
+import type {Options, Options_} from './types';
```
Comment (jaredwray) [inline packages/tiered/src/index.ts:3]: What is `Options_` vs `Options` ?

File: packages/tiered/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+		this.opts = {
+			validator: () => true,
+			dialect: 'tiered',
+			...options,
+		};
+		this.remote = remote;
+		this.local = local;
+	}
+
+	async get(key: string): Promise<any> {
```
Comment (jaredwray) [inline packages/tiered/src/index.ts:24]: Do we want to return `any` or could we define this more?

**Review by jaredwray — CHANGES_REQUESTED** [review]

File: packages/tiered/src/index.ts (hunk 1 above)
Comment (chrisllontop) [reply packages/tiered/src/index.ts:3]: Options is the parameters that we receive in the constructor. Options_ are the parameters in this.opts

File: packages/tiered/src/index.ts (hunk 2 above)
Comment (chrisllontop) [reply packages/tiered/src/index.ts:24]: We don't know the data type cached, for this reason, we can return any.

**Review by jaredwray — APPROVED** [review]

**Pull request merged**